From 6028fa644cba326b1fd80a52552ec020912001ef Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@pilppa.org>
Date: Tue, 14 May 2024 11:15:04 -0700
Subject: [PATCH 2/2] add more debug for error cases

- add debug in case that support for the used
  gpu is not enabled/build in to system.
  This is useful for getting better trace
  especially when trying to use
  rocm/cuda from pytorch.

Signed-off-by: Mika Laitio <lamikr@pilppa.org>
---
 hipamd/src/hip_fatbin.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hipamd/src/hip_fatbin.cpp b/hipamd/src/hip_fatbin.cpp
index dc76883f2..aa7dd285c 100644
--- a/hipamd/src/hip_fatbin.cpp
+++ b/hipamd/src/hip_fatbin.cpp
@@ -253,11 +253,13 @@ hipError_t FatBinaryInfo::ExtractFatBinaryUsingCOMGR(const std::vector<hip::Devi
       auto dev_it = unique_isa_names.find(device_name);
       // If the size is 0, then COMGR API could not find the CO for this GPU device/ISA
       if (dev_it->second.first == 0) {
+        printf("hip_fatbin.cpp: COMGR API could not find the CO for this GPU device/ISA: %s\n", device_name.c_str());
         LogPrintfError("Cannot find CO in the bundle for ISA: %s \n", device_name.c_str());
         hip_status = hipErrorNoBinaryForGpu;
         ListAllDeviceWithNoCOFromBundle(unique_isa_names);
         break;
       }
+      #printf("hip_fatbin.cpp: COMGR API found CO for this GPU device/ISA: %s\n", device_name.c_str());
       guarantee(unique_isa_names.cend() != dev_it,
                 "Cannot find the device name in the unique device name");
       fatbin_dev_info_[devices[dev_idx]->deviceId()]
-- 
2.41.0

